frontend/src/components/pages/roles/RoleUpdatePage.tsx

```diff
@@ -0,0 +1,214 @@
+/**
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file https://github.com/redpanda-data/redpanda/blob/dev/licenses/bsl.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+import { createConnectQueryKey, useMutation, useQuery } from '@connectrpc/connect-query';
+import { useToast } from '@redpanda-data/ui';
+import { useQueryClient } from '@tanstack/react-query';
+import { useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import {
+  convertRulesToCreateACLRequests,
+  getAclFromAclListResponse,
+  getIdFromCreateACLRequest,
+  getOperationsForResourceType,
+  ModeAllowAll,
+  ModeDenyAll,
+  OperationTypeAllow,
+  OperationTypeDeny,
+  PrincipalTypeRedpandaRole,
+  type Rule,
+  type SharedConfig,
+} from 'components/pages/acls/new-acl/ACL.model';
+import CreateACL from 'components/pages/acls/new-acl/CreateACL';
+import {
+  ACLService,
+  type CreateACLRequest,
+  type DeleteACLsRequest,
+  type ListACLsRequest,
+} from 'protogen/redpanda/api/dataplane/v1/acl_pb';
+import { createACL, deleteACLs, listACLs } from 'protogen/redpanda/api/dataplane/v1/acl-ACLService_connectquery';
+import { uiState } from '../../../state/uiState';
+import PageContent from '../../misc/PageContent';
+
+interface ACLWithId extends CreateACLRequest {
+  id: string;
+}
+
+interface ACLDifference {
+  toCreate: ACLWithId[];
+  toDelete: ACLWithId[];
+}
+
+/**
+ * Compares current ACL rules with new rules to determine which need to be created and deleted
+ * @param currentRules - The existing ACL rules
+ * @param newRules - The desired ACL rules
+ * @returns Object containing arrays of rules to create and delete
+ */
+function calculateACLDifference(currentRules: ACLWithId[], newRules: ACLWithId[]): ACLDifference {
+  const currentIds = new Set(currentRules.map((r) => r.id));
+  const newIds = new Set(newRules.map((r) => r.id));
+
+  // Rules to create: in newRules but not in currentRules
+  const toCreate = newRules.filter((rule) => !currentIds.has(rule.id));
+
+  // Rules to delete: in currentRules but not in newRules
+  const toDelete = currentRules.filter((rule) => !newIds.has(rule.id));
+
+  return {
+    toCreate,
+    toDelete,
+  };
+}
+
+const RoleUpdatePage = () => {
+  const toast = useToast();
+  const nav = useNavigate();
+  const queryClient = useQueryClient();
+  const { roleName = '' } = useParams<{ roleName: string }>();
+
+  useEffect(() => {
+    uiState.pageBreadcrumbs = [
+      { title: 'Security', linkTo: '/security' },
+      { title: 'Roles', linkTo: '/security/roles' },
+      { title: roleName, linkTo: `/security/roles/${roleName}/details` },
+      { title: 'Update Role', linkTo: '', heading: '' },
+    ];
+  }, [roleName]);
```

Comment:
This should be moved to `runInAction()` MobX util method, which should be ran in a `useEffect()` you have now, but you could do something like
```
// Hack for MobX to ensure we don't need to use observables
export const updatePageTitle = () => {
  runInAction(() => {
    uiState.pageTitle = 'Remote MCP';
    uiState.pageBreadcrumbs.pop(); // Remove last breadcrumb to ensure the title is used without previous page breadcrumb being shown
    uiState.pageBreadcrumbs.push({ title: 'Remote MCP', linkTo: '/remote-mcp', heading: 'Remote MCP' });
  });
};
```

Reply:
let's avoid to use `Mobx` we need to remove it, therfore not use on new components